test/test_other.py

```diff
@@ -1273,6 +1273,10 @@ def test_wl_stackfirst(self):
     err = self.expect_fail(cmd + ['-sGLOBAL_BASE=1024'])
     self.assertContained('error: --stack-first is not compatible with -sGLOBAL_BASE', err)
 
+  def test_dylink_global_base(self):
+    err = self.expect_fail([EMCC, test_file('hello_world.c'), '-sGLOBAL_BASE=1024', '-sMAIN_MODULE'])
+    self.assertContained('emcc: error: GLOBAL_BASE is not compatible with relocatable output', err)
+
```

Comment:
Testing a side module instead would be faster, and also be future compatible if we ever make main modules non-relocatable. But just a suggestion, lgtm either way.

Reply:
The error happens before we do any actual linking work so its just as fast either way.